src/main/scala/codacy/plugins/test/MultipleTests.scala

```diff
@@ -0,0 +1,143 @@
+package codacy.plugins.test
+
+import java.nio.file.Path
+
+import codacy.utils.Printer
+import com.codacy.analysis.core.model.{
+  CodacyCfg,
+  Configuration,
+  FileCfg,
+  FileError,
+  FullLocation,
+  Issue,
+  LineLocation,
+  Parameter,
+  Pattern
+}
+import com.codacy.analysis.core.tools.Tool
+import com.codacy.plugins.api.results.Result
+import com.codacy.plugins.api.results.Result.Level
+import com.codacy.plugins.results.traits.{DockerToolDocumentation, ToolRunner}
+import com.codacy.plugins.runners.{BinaryDockerRunner, DockerRunner}
+import com.codacy.plugins.utils.BinaryDockerHelper
+import com.codacy.plugins.results.PluginResult
+
+import better.files._
+
+import play.api.libs.json.Json
+
+import scala.util.{Failure, Success, Try}
+import com.fasterxml.jackson.core.JsonParseException
+import scala.xml.XML
+
+object MultipleTests extends ITest with CustomMatchers {
```

Comment:
this file contains way too much code and have multiple responsibilities.
I see at least:
 - tests detection
 - data collection for preparing the tests to run
 - actual execution of the configured tests
 - emitting data in the checkstyle format and various parsing

Split into separate classes for easier testability/readability please